From fdf24f98bc9e4632bfe7f280872cbbf40942ae7b Mon Sep 17 00:00:00 2001
From: Colin Watson <cjwatson@ubuntu.com>
Date: Mon, 31 Mar 2014 13:51:17 +0100
Subject: [PATCH 069/143] Fix grub-probe -0 option

* util/grub-probe,c (options): Make -0 work again (broken by
conversion to argp).
(main): Simplify logic.
---
 ChangeLog         | 6 ++++++
 util/grub-probe.c | 7 ++-----
 2 files changed, 8 insertions(+), 5 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 9404eda..efbed8c 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,9 @@
+2014-03-31  Colin Watson  <cjwatson@ubuntu.com>
+
+	* util/grub-probe,c (options): Make -0 work again (broken by
+	conversion to argp).
+	(main): Simplify logic.
+
 2014-03-26  Vladimir Serbinenko  <phcoder@gmail.com>
 
 	* grub-core/lib/relocator.c: Fix the case when end of leftover is used.
diff --git a/util/grub-probe.c b/util/grub-probe.c
index 1f3b59f..80509be 100644
--- a/util/grub-probe.c
+++ b/util/grub-probe.c
@@ -711,6 +711,7 @@ static struct argp_option options[] = {
    N_("use FILE as the device map [default=%s]"), 0},
   {"target",  't', N_("TARGET"), 0, 0, 0},
   {"verbose",     'v', 0,      0, N_("print verbose messages."), 0},
+  {0, '0', 0, 0, N_("separate items in output using ASCII NUL characters"), 0},
   { 0, 0, 0, 0, 0, 0 }
 };
 
@@ -884,11 +885,7 @@ main (int argc, char *argv[])
   else
     probe (arguments.devices[0], NULL, delim);
 
-  if (!arguments.zero_delim && (print == PRINT_BIOS_HINT
-				|| print == PRINT_IEEE1275_HINT
-				|| print == PRINT_BAREMETAL_HINT
-				|| print == PRINT_EFI_HINT
-				|| print == PRINT_ARC_HINT))
+  if (delim == ' ')
     putchar ('\n');
 
   /* Free resources.  */
-- 
1.9.3

